Topic1/src/com/topic1/CompositeOffer.java

```diff
@@ -0,0 +1,18 @@
+package com.topic1;
+
+public class CompositeOffer implements Offer {
+
+	Offer offer;
+
+	public CompositeOffer(Offer offer) {
+
+		this.offer = offer;
+
+	}
+
+	public void displayOffer() {
+
+		offer.displayOffer();
```

Comment:
[High] For a composite Offer, you should iterate over all elements and display the message.